multi/nrf91-multi/uart.c

```diff
@@ -0,0 +1,536 @@
+/*
+ * Phoenix-RTOS
+ *
+ * nRF91 UART driver
+ *
+ * Copyright 2023, 2024 Phoenix Systems
+ * Author: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <paths.h>
+#include <posix/utils.h>
+#include <sys/msg.h>
+#include <sys/pwman.h>
+#include <sys/interrupt.h>
+#include <sys/file.h>
+#include <sys/threads.h>
+#include <sys/types.h>
+#include <libtty.h>
+#include <libtty-lf-fifo.h>
+
+#include <board_config.h>
+
+#include "nrf91-multi.h"
+#include "common.h"
+#include "uart.h"
+
+#define UART_CNT (UART0 + UART1 + UART2 + UART3)
+
+#define THREADS_CNT     3
+#define THREAD_STACKSZ  768
+#define THREAD_PRIORITY 1
+
+typedef struct {
+	char stack[512] __attribute__((aligned(8)));
+
+	volatile unsigned int *base;
+	volatile char *txDma;
+	volatile char *rxDma;
+
+	unsigned char stopBits;
+	int parity;
+	int baud;
+	int cnt;
+	bool init;
+	bool resetCnt;
+
+	handle_t cond;
+	handle_t irqlock;
+
+	libtty_common_t ttyCommon;
+
+	volatile bool rxdrdy;
+	int rxdCount;
```

Comment:
Changed in an irq handler
```suggestion
	volatile int rxdCount;
```

Reply:
Oh, that's right, thanks!